lib/Dialect/Moore/MooreOps.cpp

```diff
@@ -265,6 +265,30 @@ VariableOp::handlePromotionComplete(const MemorySlot &slot, Value defaultValue,
   return std::nullopt;
 }
 
+LogicalResult VariableOp::canonicalize(VariableOp op,
+                                       ::mlir::PatternRewriter &rewriter) {
+  llvm::DenseSet<Operation *> variables;
+  Value initial;
+  for (auto *user : op->getUsers())
+    if (isa<ContinuousAssignOp>(user) &&
+        (user->getOperand(0) == op.getResult())) {
+      // Don't canonicalize the multiple continuous assignment to the same
+      // variable.
+      if (variables.contains(op))
+        return failure();
+
+      variables.insert(op);
+      initial = user->getOperand(1);
+    }
```

Comment:
I think we don't need `variables` set.
```suggestion
  Value initial;
  for (auto *user : op->getUsers())
    if (isa<ContinuousAssignOp>(user) &&
        (user->getOperand(0) == op.getResult())) {
      // Don't canonicalize the multiple continuous assignment to the same
      // variable.
      if (initial)
        return failure();

      initial = user->getOperand(1);
    }
```